tests/foundry/foundry.k.check.expected

```diff
@@ -3016,64 +3015,157 @@ module LOOPSTEST-BIN-RUNTIME
+      
+    
+    rule  ( selector ( "testSortBroken(uint256[])" ) => 3541275661 )
+      
+    
+    rule  ( selector ( "testSqrt(uint256)" ) => 233106198 )
       
     
-    rule  ( selector ( "IS_TEST" ) => 4202047188 )
+    rule  ( selector ( "testSumToN(uint256)" ) => 2289979355 )
       
     
-    rule  ( selector ( "failed" ) => 3124842406 )
+    rule  ( selector ( "testSumToNBroken(uint256)" ) => 1318374999 )
       
     
-    rule  ( selector ( "setUp" ) => 177362148 )
+    rule  ( selector ( "test_sum_10()" ) => 1087009050 )
       
     
-    rule  ( selector ( "testIsNotPrime" ) => 809987623 )
+    rule  ( selector ( "test_sum_100()" ) => 1519953344 )
       
     
-    rule  ( selector ( "testIsPrime" ) => 111940912 )
+    rule  ( selector ( "test_sum_1000()" ) => 222767225 )
       
     
-    rule  ( selector ( "testIsPrimeBroken" ) => 1834826207 )
+    rule  ( selector ( "vm()" ) => 980845667 )
       
+
+endmodule
+
+module METHODDISAMBIGUATETEST-BIN-RUNTIME
+    imports public FOUNDRY
+    
+    syntax Contract ::= MethodDisambiguateTestContract
+    
+    syntax MethodDisambiguateTestContract ::= "MethodDisambiguateTest" [symbol(), klabel(contract_MethodDisambiguateTest)]
+    
+      
+    
+    syntax Field ::= MethodDisambiguateTestField
+    
+    syntax MethodDisambiguateTestField ::= "IS_TEST" [symbol(), klabel(field_MethodDisambiguateTest_IS_TEST)]
+    
+    syntax MethodDisambiguateTestField ::= "_failed" [symbol(), klabel(field_MethodDisambiguateTest__failed)]
+    
+    syntax MethodDisambiguateTestField ::= "IS_SCRIPT" [symbol(), klabel(field_MethodDisambiguateTest_IS_SCRIPT)]
+    
+    syntax MethodDisambiguateTestField ::= "stdstore" [symbol(), klabel(field_MethodDisambiguateTest_stdstore)]
+    
+    rule  ( #loc ( MethodDisambiguateTest . IS_TEST ) => 0 )
+      
+    
+    rule  ( #loc ( MethodDisambiguateTest . _failed ) => 0 )
+      
+    
+    rule  ( #loc ( MethodDisambiguateTest . IS_SCRIPT ) => 0 )
+      
+    
+    rule  ( #loc ( MethodDisambiguateTest . stdstore ) => 1 )
+      
+    
+    syntax Bytes ::= MethodDisambiguateTestContract "." MethodDisambiguateTestMethod [function(), symbol(), klabel(method_MethodDisambiguateTest)]
+    
+    syntax MethodDisambiguateTestMethod ::= "IS_SCRIPT" "(" ")" [symbol(), klabel(method_MethodDisambiguateTest_IS_SCRIPT_)]
+    
+    syntax MethodDisambiguateTestMethod ::= "IS_TEST" "(" ")" [symbol(), klabel(method_MethodDisambiguateTest_IS_TEST_)]
     
-    rule  ( selector ( "testIsPrimeOpt" ) => 404719852 )
+    syntax MethodDisambiguateTestMethod ::= "failed" "(" ")" [symbol(), klabel(method_MethodDisambiguateTest_failed_)]
+    
+    syntax MethodDisambiguateTestMethod ::= "getNumber" "(" Int ")" [symbol(), klabel(method_MethodDisambiguateTest_getNumber_Int)]
+    
+    syntax MethodDisambiguateTestMethod ::= "getNumber" "(" Int ")" [symbol(), klabel(method_MethodDisambiguateTest_getNumber_Int)]
```

Comment:
It looks like they're getting the same klabels.....

And they have the same production. So we need to use the solidity type in the KLabel, not the K sort.

Also, for methods that are overloaded like this, I think we will just have to include the types directly in the production. We could instead generate, for example:

```
syntax MethodDisambiguateTestMethod ::= "getNumber" "(" Int ":uint32" ")" [symbol(), klabel(method_MethodDisambiguateTest_getNumber_uint32]
```

Reply:
So that it looks kinda like solidity, but we have the `:uint32` to disambiguate the types present in teh production.

Reply:
I've added it so that the solidity types are included in the productions, as suggested.